jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -260,22 +263,40 @@ private Image pullBaseImage(
   }
 
   /**
-   * Looks through a manifest list for any amd64/linux manifest and downloads and returns the first
-   * manifest it finds.
+   * Looks through a manifest list for the user specified arch/os manifest and downloads and returns
+   * the first manifest it finds.
    */
-  private ManifestAndDigest<?> obtainPlatformSpecificImageManifest(
+  @VisibleForTesting
+  ManifestAndDigest<?> obtainPlatformSpecificImageManifest(
       RegistryClient registryClient, V22ManifestListTemplate manifestListTemplate)
       throws RegistryException, IOException {
 
-    List<String> digests = manifestListTemplate.getDigestsForPlatform("amd64", "linux");
+    Set<Platform> platforms = buildContext.getContainerConfiguration().getPlatforms();
+    Platform platform = platforms.iterator().next();
+    String architecture = platform.getArchitecture();
+    String os = platform.getOs();
+
+    buildContext
+        .getEventHandlers()
+        .dispatch(
```

Comment:
```suggestion
    EventHandlers eventHandlers = buildContext.getEventHandlers();
    eventHandlers.dispatch(
```

Then on line 300, you can reuse `eventHandlers`.